File: python/bob/bob.py
def response(hey_bob):

    hey_bob = hey_bob.strip()

    if hey_bob == "":
        return "Fine. Be that way!"

    if is_yelling(hey_bob) and is_question(hey_bob):
        return "Calm down, I know what I'm doing!"

    if is_yelling(hey_bob):
        return "Whoa, chill out!"

    if is_question(hey_bob):
        return "Sure."

    return "Whatever."


def is_question(phrase):
    return phrase.endswith("?")

def is_yelling(phrase):
    return phrase.isupper()

File: fsharp/bob/Bob.fs
module Bob

open System
open System.Text.RegularExpressions

let response (input: string): string = 
    let str = input.Trim()
    let isQuestion (input: string): bool = Regex.IsMatch(input, "^.*\?$")
    let isYelling (input: string): bool = (input = input.ToUpper()) && not (input = input.ToLower())

    match (str, isYelling(str), isQuestion(str)) with
    | ("", _, _) -> "Fine. Be that way!"
    | (_, true, true) -> "Calm down, I know what I'm doing!"
    | (_, true, false) -> "Whoa, chill out!"
    | (_, false, true) -> "Sure."
    | _ -> "Whatever."
